engine/table/src/main/java/io/deephaven/engine/table/impl/perf/PerformanceEntry.java

```diff
@@ -50,6 +50,9 @@ public class PerformanceEntry extends BasePerformanceEntry implements TableListe
         this.description = description;
         this.callerLine = callerLine;
         authContext = id == QueryConstants.NULL_INT ? null : ExecutionContext.getContext().getAuthContext();
+        if (id != QueryConstants.NULL_INT && authContext == null) {
+            throw new IllegalStateException("No Auth Context is Installed. Do you need to open an ExecutionContext?");
```

Comment:
Do we think there is a path that gets us here besides engine bugs and thread mis-use? What AuthContext do clients get by default?
```suggestion
            throw new IllegalStateException("No AuthContext is installed. Do you need to open an ExecutionContext?");
```

Reply:
A user could see this by constructing a listener on a new thread.